satpy/readers/olci_nc.py

```diff
@@ -100,30 +101,40 @@ def __init__(self, filename, filename_info, filetype_info,
         """Init the olci reader base."""
         super(NCOLCIBase, self).__init__(filename, filename_info,
                                          filetype_info)
-        self.nc = xr.open_dataset(self.filename,
+        self._engine = engine
+        self._start_time = filename_info['start_time']
+        self._end_time = filename_info['end_time']
+        # TODO: get metadata from the manifest file (xfdumanifest.xml)
+        self.platform_name = PLATFORM_NAMES[filename_info['mission_id']]
+        self.sensor = 'olci'
+        self.open_file = None
+
+    @property
+    @lru_cache(maxsize=2)
+    def nc(self):
```

Comment:
Mostly for my curiosity: I've never seen `lru_cache` used on a `property` before.  I don't know much about OLCI.  What is the purpose and effect of a cache of size 2 on this property?  Does that mean that if I open 3 OLCI files that the one I opened first is closed again?  Or just that there is work redone if `.nc` is accessed on the first one?

Reply:
I put size 2 for safety, but in principle only 1 is needed, as this is a property on the instance. So each instance of the file handler will have a different cache for this accessor, and allows repeated calls to the property not to trigger a reread (especially important when this is a remote file of course).

Reply:
Wasn't this cached_property before? What's the difference?

Reply:
it was, but `cached_property` was introduced in python 3.8, so I have to pass on that for now!

Reply:
Ah ok. What about a try/except ImportError on that? Can lru_cache take a maxsize=1? Does it default to that? Basically can we use cached_property when it exists, but use lru_cache for backwards compatibility for now?

Reply:
I can do that.

Reply:
Hmmm, I thought this would do it, but no... Will investigate more tomorrow
```python
def cached_property(func):
    return property(lru_cache(maxsize=1)(func))
```

Reply:
ok, got it to work now.